### PR TITLE
added sha256_secrethash

### PR DESCRIPTION
**Pull request by hackaugusto**
This isolates the hash function, the main goal here is to avoid getting in the habit of casting everywhere.

### CODE REVIEW

File: raiden/utils/secrethash.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from hashlib import sha256
+
+from raiden.utils.typing import Secret, SecretHash
+
+
+def sha256_secrethash(secret: Secret) -> SecretHash:
+    """Compute the secret hash using sha256."""
+    return SecretHash(sha256(secret).digest())
```
Comment (palango) [inline raiden/utils/secrethash.py]: Seems simple enough, but a simple test would be nice.

File: raiden/tests/utils/transport.py [hunk 2]
```diff
@@ -160,11 +157,7 @@ def generate_config(port: int):
             )
         return server_name, config_file
 
-    try:
-        yield generate_config
-    finally:
-        if delete_base_dir:
-            shutil.rmtree(synapse_base_dir)
```
Comment (palango) [inline raiden/tests/utils/transport.py]: - How is this related to this PR?
- Don't we need to clean up the `synapse_base_dir` any more?

Comment (hackaugusto) [reply raiden/tests/utils/transport.py]: Synapse failed to run because of a import cycle. Having the file was useful to actually find the problem, since the error message was being redirected to devnull. I can remove it if you want.

Comment (palango) [reply raiden/tests/utils/transport.py]: No strong feelings on my side. If you think we should remove it that's fine.

File: raiden/utils/secrethash.py (hunk 1 above)
Comment (palango) [reply raiden/utils/secrethash.py]: Can you add a small test for this?

Comment (hackaugusto) [reply raiden/utils/secrethash.py]: what do you want me to test? the downstream library?

Comment (hackaugusto) [reply raiden/utils/secrethash.py]: done
